isis/tests/FunctionalTestsHiCubeNorm.cpp

```diff
@@ -0,0 +1,216 @@
+  }
+
+  Cube oCube(outCubeFileName, "r"); 
+  oCube.reopen("rw");
+
+  Histogram *oCubeStats = oCube.histogram();
+
+  ASSERT_DOUBLE_EQ(oCubeStats->Average(), -1462164.1207275416); 
+  ASSERT_DOUBLE_EQ(oCubeStats->Sum(), -78746310885.902481); 
+  ASSERT_EQ(oCubeStats->ValidPixels(), 53856);  
+  ASSERT_DOUBLE_EQ(oCubeStats->StandardDeviation(), 9547737.4830061328);  
+
+  delete oCubeStats;
+}
+
+
+TEST_F(MroCube, FunctionalTestsHiCubeNormAverage) {
+  setInstrument("-74999", "HIRISE", "MARS RECONNAISSANCE ORBITER");
+
+  QTemporaryDir prefix;
+  QString outCubeFileName = prefix.path() + "/outTEMP.cub";
+  QVector<QString> args = {"to="+outCubeFileName, "mode=subtract", "normalizer=average"};
+  
+  UserInterface options(APP_XML, args);
+  try {
+    hicubenorm(testCube, options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to process HRISE image: " << e.what() << std::endl;
+  }
+
+  Cube oCube(outCubeFileName, "r"); 
+  oCube.reopen("rw");
+
+  Histogram *oCubeStats = oCube.histogram();
+  
+  ASSERT_DOUBLE_EQ(oCubeStats->Average(), 602.49999332903235); 
+  ASSERT_DOUBLE_EQ(oCubeStats->Sum(), 766032951.51837158); 
+  ASSERT_EQ(oCubeStats->ValidPixels(), 1271424);  
+  ASSERT_DOUBLE_EQ(oCubeStats->StandardDeviation(), 347.47625973541767);  
+
+  delete oCubeStats;
+}
+
+
+TEST_F(MroCube, FunctionalTestsHiCubeNormNewVersion) {
+  setInstrument("-74999", "HIRISE", "MARS RECONNAISSANCE ORBITER");
+
+  QTemporaryDir prefix;
+  QString tablePath = "/tmp/stats.pvl"; // prefix.path() + "/stats.pvl"; 
+  QString outCubeFileName = prefix.path() + "/outTEMP.cub";
+
+  // Generate a Statistics file
+  PvlGroup stats("Results");
+  for(int i = 1; i <= 1204; i++) {      
+    stats += PvlKeyword("Band", "1");
+    stats += PvlKeyword("RowCol", QString::number(i));
+    stats += PvlKeyword("ValidPixels", QString::number(1056));
+    stats += PvlKeyword("Mean", QString::number(i));
+    stats += PvlKeyword("Median", QString::number(i/2));
+    // the rest shouldn't matter
+    stats += PvlKeyword("Std", "0.0");     
+    stats += PvlKeyword("Minimum", "0.0");
+    stats += PvlKeyword("Maximum", "0.0");
+  }
+  
+  Pvl table; 
+  table.addGroup(stats);
+  table.write(tablePath);
+
+  QVector<QString> args = {"to="+outCubeFileName, "mode=subtract", "fromstats="+tablePath, "statsource=pvl", "normalizer=average", "new_version=yes"};
+  
+  UserInterface options(APP_XML, args);
+  try {
+    hicubenorm(testCube, options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to process HRISE image: " << e.what() << std::endl;
+  }
+
+  Cube oCube(outCubeFileName, "r"); 
+  oCube.reopen("rw");
+
+  Histogram *oCubeStats = oCube.histogram();
+  
+  ASSERT_DOUBLE_EQ(oCubeStats->Average(), 602.50000120516233); 
+  ASSERT_DOUBLE_EQ(oCubeStats->Sum(), 766032961.53227234); 
+  ASSERT_EQ(oCubeStats->ValidPixels(), 1271424);  
+  ASSERT_DOUBLE_EQ(oCubeStats->StandardDeviation(), 347.55180258334673);  
```

Comment:
Wasn't sure how to best test DNs, using stats seems like a decent answer?

Reply:
Without testing individual DNs, stats should be a good compromise.